### PR TITLE
Publish docs to Nexus 3 using its REST API

### PR DESCRIPTION
**Pull request by cwardgar**
* Add PublishToRawRepoTask, which publishes all files in a given directory to a raw repository. Also add unit tests, which assert correct resource path creation.
* Add DeleteFromNexusTask, which deletes components from Nexus that match a query. This is mostly for repo maintenance; mass deletions are tedious otherwise.
* Add 'docs' subproject, which has an instance of each task. This will be where the new Kramdown version of our docs go, when they're ready.
* Move PublishingUtil.groovy and its test into the 'edu.ucar.build.publishing' package, where they logically belong.
* Add "http-builder-ng-core" dependency, which has a spiffy Groovy DSL for making REST calls.

### CODE REVIEW

**Comment by lesserwhirls** [comment]
Sorry about that @cwardgar - my comment was listed as pending and was invisible until I submitted it.

File: docs/build.gradle [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import edu.ucar.build.publishing.DeleteFromNexusTask
+import edu.ucar.build.publishing.PublishToRawRepoTask
+
+apply from: "$rootDir/gradle/any/properties.gradle"  // For Nexus credential properties.
+
+task publish(type: PublishToRawRepoTask) {
+    // TODO: Move these properties to any/publishing.gradle.
+    host = "https://artifacts.unidata.ucar.edu/"
+    repoName = "thredds-doc"
```
[the remark this answers is not in the document]
Comment (cwardgar) [reply docs/build.gradle]: Yep.

**Comment by cwardgar** [comment]
No sweat. I'll leave this to you to merge, if you have no edits to propose.

**Pull request merged**